.ci/compute_projects.py

```diff
@@ -0,0 +1,186 @@
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Computes the list of projects that need to be tested from a diff.
+
+Does some things, spits out a list of projects.
+"""
+
+from collections.abc import Set
+import pathlib
+import platform
+import sys
+
+PROJECT_DEPENDENCIES = {
+    "llvm": set(),
+    "clang": {"llvm"},
+    "bolt": {"clang", "lld", "llvm"},
+    "clang-tools-extra": {"clang", "llvm"},
+    "compiler-rt": {"clang", "lld"},
+    "libc": {"clang", "lld"},
+    "openmp": {"clang", "lld"},
+    "flang": {"llvm", "clang"},
+    "lldb": {"llvm", "clang"},
+    "libclc": {"llvm", "clang"},
+    "lld": {"llvm"},
+    "mlir": {"llvm"},
+    "polly": {"llvm"},
+}
+
+DEPENDENTS_TO_TEST = {
+    "llvm": {"bolt", "clang", "clang-tools-extra", "lld", "lldb", "mlir", "polly"},
+    "lld": {"bolt", "cross-project-tests"},
+    "clang": {"clang-tools-extra", "compiler-rt", "cross-project-tests"},
+    "clang-tools-extra": {"libc"},
+    "mlir": {"flang"},
+}
+
+DEPENDENT_RUNTIMES_TO_TEST = {"clang": {"libcxx", "libcxxabi", "libunwind"}}
+
+EXCLUDE_LINUX = {
+    "cross-project-tests",  # Tests are failing.
```

Comment:
This and others below, I think we should require GitHub issues for each exclusion item.

Reply:
Well, each one that is excluded because of an engineering problem. If something isn't relevant, a comment should be sufficient.

Reply:
Good point. I opened issues for all the exclusions, minus the projects that are just not supported on windows.